src/index.js

```diff
@@ -4,556 +4,608 @@ const codec = require('./codec')
-    case 'IPv4':
-      ip = 'ip4'
-      break
-    case 'IPv6':
-      ip = 'ip6'
-      break
-    default:
-      throw Error(`Invalid addr family. Got '${addr.family}' instead of 'IPv4' or 'IPv6'`)
+  /**
+   * Creates a Multiaddr from a node-friendly address object
+   *
+   * @param {{family: string, address: string, port: string}} addr
+   * @param {string} transport
+   * @returns {Multiaddr} multiaddr
+   * @throws {Error} Throws error if addr or transport are not truthy
+   * @example
+   *
+   * ```js
+   * Multiaddr.fromNodeAddress({address: '127.0.0.1', port: '4001'}, 'tcp')
+   * // <Multiaddr 047f000001060fa1 - /ip4/127.0.0.1/tcp/4001>
+   * ```
+   */
+  static fromNodeAddress (addr, transport) {
+    if (!addr) { throw new Error('requires node address object') }
+    if (!transport) { throw new Error('requires transport protocol') }
+    let ip
+    switch (addr.family) {
+      case 'IPv4':
+        ip = 'ip4'
+        break
+      case 'IPv6':
+        ip = 'ip6'
+        break
+      default:
+        throw Error(`Invalid addr family. Got '${addr.family}' instead of 'IPv4' or 'IPv6'`)
+    }
+    return new Multiaddr('/' + [ip, addr.address, transport, addr.port].join('/'))
   }
-  return Multiaddr('/' + [ip, addr.address, transport, addr.port].join('/'))
-}
 
-// TODO find a better example, not sure about it's good enough
-/**
- * Returns if a Multiaddr is a Thin Waist address or not.
- *
- * Thin Waist is if a Multiaddr adheres to the standard combination of:
- *
- * `{IPv4, IPv6}/{TCP, UDP}`
- *
- * @param {Multiaddr} [addr] - Defaults to using `this` instance
- * @returns {boolean} isThinWaistAddress
- * @example
- * const mh1 = Multiaddr('/ip4/127.0.0.1/tcp/4001')
- * // <Multiaddr 047f000001060fa1 - /ip4/127.0.0.1/tcp/4001>
- * const mh2 = Multiaddr('/ip4/192.168.2.1/tcp/5001')
- * // <Multiaddr 04c0a80201061389 - /ip4/192.168.2.1/tcp/5001>
- * const mh3 = mh1.encapsulate(mh2)
- * // <Multiaddr 047f000001060fa104c0a80201061389 - /ip4/127.0.0.1/tcp/4001/ip4/192.168.2.1/tcp/5001>
- * mh1.isThinWaistAddress()
- * // true
- * mh2.isThinWaistAddress()
- * // true
- * mh3.isThinWaistAddress()
- * // false
- */
-Multiaddr.prototype.isThinWaistAddress = function isThinWaistAddress (addr) {
-  const protos = (addr || this).protos()
+  /**
+   * Returns if something is a Multiaddr that is a name
+   *
+   * @param {Multiaddr} addr
+   * @returns {boolean} isName
+   */
+  static isName (addr) {
+    if (!Multiaddr.isMultiaddr(addr)) {
+      return false
+    }
 
-  if (protos.length !== 2) {
-    return false
+    // if a part of the multiaddr is resolvable, then return true
+    return addr.protos().some((proto) => proto.resolvable)
   }
 
-  if (protos[0].code !== 4 && protos[0].code !== 41) {
-    return false
+  /**
+   * Returns an array of multiaddrs, by resolving the multiaddr that is a name
+   *
+   * @async
+   * @param {Multiaddr} addr
+   * @returns {Promise<Multiaddr[]>}
+   */
+  static resolve (addr) {
```

Comment:
This can be removed right? I did not see this when I implemented the `multiaddr.resolve()` and I think we can just leverage that.

Reply:
yes